hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/MapIndexScanP.java

```diff
@@ -149,9 +150,22 @@ public boolean complete() {
         return isIndexSorted ? runSortedIndex() : runHashIndex();
     }
 
+    /**
+     * Processor uses operation runner, which may hold locks
+     * for {@link java.util.concurrent.ConcurrentHashMap#computeIfAbsent(Object, Function)}.
+     * <p>
+     * See more : {@link com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl#run(Operation, long)}
+     *
+     * @return false
+     */
+    @Override
+    public boolean isCooperative() {
+        return false;
+    }
+
     @Override
     public boolean closeIsCooperative() {
-        return true;
+        return false;
```

Comment:
`closeIsCooperative()` is ignored when `isCooperative()` is false and `false` is default for `closeIsCooperative` so implementation of this method can be removed from here

Reply:
I prefer explicit definitions, so, I'd stay with that method

Reply:
This method should return `true`. The `close()` does nothing, therefore we don't need to offload the call to it to another thread. You should read the `closeIsCooperative()` javadoc.